sdks/go/node/core/parallelCaller.go

```diff
@@ -85,22 +84,21 @@ func (pc _parallelCaller) Call(
 
 		childCallIndexByID[childCallID] = childCallIndex
 
+		// for "needs" children, track a specific context so we can kill that child
+		// use the normal parallel context for other children
+		var childCtx context.Context
 		if childCall.Name != nil {
-			childCallIDByName[*childCall.Name] = childCallID
+			childCallIndexByName[*childCall.Name] = childCallIndex
+			childCtx, childCallCancellationByIndex[childCallIndex] = context.WithCancel(parallelCtx)
+		} else {
+			childCtx = parallelCtx
 		}
 
+		wg.Add(1)
 		go func(childCall *model.CallSpec) {
-			defer func() {
```

Comment:
The reason we had this was, uncaught panics inside  go routines actually crash your application and we don't want that. I think we need to keep this bit.

Reply:
This feels a pattern that will swallow errors instead of giving us the chance to fix them. This doesn't actually expose the reason for the panic in the same manner normal errors are exposed. I generally prefer to allow panics to crash the program, so there's a more obvious failure to investigate.

That being said, now that errors are returned through the `childResults` `chan`, I can return panics through that same mechanism which cleans this up a lot.

Reply:
Either way I think it would be a nice enhancement not to swallow such an error and make sure it get's returned. Plenty of options as far as how we could accomplish that.

Reply:
Yes, I made a change to explicitly return an error here